spiderfoot/correlation.py

```diff
@@ -56,33 +70,54 @@ def __init__(self, dbh: SpiderFootDb, ruleset: dict, scanId: str = None):
             try:
                 self.rules.append(yaml.safe_load(ruleset[rule_id]))
                 self.rules[len(self.rules) - 1]['rawYaml'] = ruleset[rule_id]
-                # Strip any extra newlines that may have creeped into meta
-                for rule in self.rules:
-                    for k in rule['meta'].keys():
-                        if isinstance(rule['meta'][k], str):
-                            rule['meta'][k] = rule['meta'][k].strip()
-                        else:
-                            rule['meta'][k] = rule[k]
-                continue
-            except BaseException as e:
+            except Exception as e:
                 raise SyntaxError(f"Unable to process a YAML correlation rule [{rule_id}]: {e}")
 
+        # Strip any extra newlines that may have creeped into meta name/description
+        for rule in self.rules:
+            for k in rule['meta'].keys():
+                if isinstance(rule['meta'][k], str):
+                    rule['meta'][k] = rule['meta'][k].strip()
+                else:
+                    rule['meta'][k] = rule[k]
+
         if not self.check_ruleset_validity(self.rules):
-            raise SyntaxError("Sanity check of correlation rules failed, aborting.")
+            raise SyntaxError("Sanity check of correlation rules failed.")
```

Comment:
Removing "aborting" as this isn't the right place to report state. This initialisation function doesn't know where it is being called from and hasn't "started" anything which could be "aborted" (yet). Also, for comparison, the `raise` above doesn't say "aborting" yet would have the same effect by raising during init.